packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -104,12 +174,97 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
+	 *
+	 * 3. If only `categories` or a `namespace` is provided, the event should match either one of them.
+	 *
+	 * 4. If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+	 *
+	 * @param event - The telemetry event to check against the filters.
+	 *
+	 * @returns boolean `true` if the event matches any filter, otherwise `false`.
+	 */
 	private doesEventMatchFilter(event: ITelemetryBaseEvent): boolean {
 		for (const filter of this.config.filtering.filters ?? []) {
-			if (filter.category === event.category) {
+			if ("namespacePattern" in filter) {
+				if (event.eventName.startsWith(filter.namespacePattern)) {
+					// Found matching namespace pattern, since filters are ordered in most specific first,
+					// this is the most specific, relevant matching filter for the event.
+
+					// By default, if no categories are defined then any category is a valid match.
+					let doesFilterCategoriesMatch = true;
+					if ("categories" in filter && filter.categories.length > 0) {
+						doesFilterCategoriesMatch = false;
+						const matchingCategory = filter.categories.find(
+							(category) => category === event.category,
+						);
+						doesFilterCategoriesMatch = matchingCategory ? true : false;
+					}
+
+					if (doesFilterCategoriesMatch) {
+						// The most specific, relevant filter matches so no need to attempt to evaluate against other filters
+						// as long as the events namespace does not match any defined namespace exception.
+						if (filter.namespacePatternExceptions !== undefined) {
+							for (const patternException of filter.namespacePatternExceptions) {
+								if (event.eventName.startsWith(patternException)) {
+									return false;
+								}
+							}
+						}
+						return true;
+					} else {
+						return false;
+					}
+				}
+			}
+			// Filter only has categories defined
+			else if ("categories" in filter && filter.categories.length > 0) {
+				const doesFilterCategoriesMatch = filter.categories.find(
+					(category) => category === event.category,
+				);
+
+				// This filter specified no namespaces but it has a category match.
+				// Since filters are ordered by most specific first, we know that no previous
+				// filters with namespaces matched so we can return true.
+				if (doesFilterCategoriesMatch) {
+					return true;
+				} else {
+					continue;
+				}
+			} else {
 				return true;
 			}
 		}
+
 		return false;
 	}
+
+	public static sortTelemetryFilters(filters: TelemetryFilter[]): TelemetryFilter[] {
+		return filters.sort((a, b) => {
+			const specificityScoreA = FluidAppInsightsLogger.getFilterSpecificityScore(a);
+			const specificityScoreB = FluidAppInsightsLogger.getFilterSpecificityScore(b);
+			return specificityScoreB - specificityScoreA;
+		});
+	}
+
+	private static getFilterSpecificityScore(filter: TelemetryFilter): number {
```

Comment:
Following up on the big comment above, I think this only needs to be a check for the namespacePattern length (coalesced to 0). I'd probably inline it when doing the sort.

Reply:
Updated to sort within the constructor using namespace length.